src/middleware.rs

```diff
@@ -0,0 +1,81 @@
+extern crate rocket;
+extern crate diesel;
+
+use self::rocket::Outcome;
+use self::rocket::http::Status;
+use self::rocket::request::{self, Request, FromRequest};
+use models::*;
+use helpers::*;
+use self::diesel::prelude::*;
+use establish_connection;
+use diesel::pg::PgConnection;
+use std::time::{Instant, UNIX_EPOCH};
+use std::time::SystemTime;
+
+#[derive(Debug)]
+pub struct Auth {
+    pub token: String,
+    pub user_id: i32,
+}
+
+fn authenticate(conn: &PgConnection, t_token: &str) -> Result<Auth, String> {
+    use schema::auth_tokens;
+    use schema::users;
+    use schema::auth_tokens::dsl::*;
+    use schema::users::dsl::*;
+
+    let auth_token = auth_tokens.filter(token.eq(t_token))
+        .load::<AuthToken>(conn).ok()
+        .unwrap();
+
+    if auth_token.len() > 0 {
+        let auth_token = auth_token.first().unwrap();
+        let now = epoch_now() as i64;
+
+        if now >= auth_token.expired_at {
+            return Err(String::from("Token has been expired"));
+        } else {
+            let _: AuthToken = diesel::update(auth_tokens.find(t_token))
+                .set(expired_at.eq(now + 30 * 24 * 60 ^ 60))
```

Comment:
These magic calculation should be a constant

Reply:
classic!

`^` 🤦‍♂️